### ISSUE TITLE
Questions about gradient values

### ISSUE THREAD

**Issue by shaoyangxu** [varun19299/rigl-reproducibility#4]
Hello, i found the grad matrix([grad = weight.grad](https://github.com/varun19299/rigl-reproducibility/blob/97443beac90e03f899652943594695e5152c2b09/sparselearning/funcs/grow.py#L86)) has many non-zero elements while their corresponding values in weight matrix are zero, I want to ask why this happen(as a beginner of pytorch)? Thanks!

partial gradient matrix:
![image](https://user-images.githubusercontent.com/59757561/168483351-e122afb4-e6c2-4663-9f39-da233086e9aa.png)

partial weight matrix:
![image](https://user-images.githubusercontent.com/59757561/168483377-f9f02e48-1c16-4312-ab08-16efb79be936.png)

**Comment by ZichenMiao** [comment]
I suppose it's because the mask is applied by changing the data (torch.tensor.data) not the tensor itself, which means the gradients of these zero values are still computed.
